test/e2e/e2e_tests_suite_test.go

```diff
@@ -0,0 +1,222 @@
+		err = utilsHelpers.CreateNamespace(ns)
+		if err != nil {
+			return err
+		}
+
+		defer func() {
+			if err := utilsHelpers.DeleteNamespace(ns); err != nil {
+				fmt.Printf("failed cleaning namespace %s: %v", ns, err)
+			}
+		}()
+
+		debugPod, err := utilsHelpers.CreatePodOnNode(node.Name, ns, consts.DefaultDebugPodImage)
+		if err != nil {
+			return fmt.Errorf("failed to create debug pod on node %s: %w", node.Name, err)
+		}
+
+		defer func() {
+			err := utilsHelpers.DeletePod(debugPod)
+			if err != nil {
+				fmt.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+			}
+		}()
+
+		command := []string{
+			"bash", "-c",
+			fmt.Sprintf("mkdir -p nft && echo '%s' > nft/firewall.nft", string(NFTtable)),
+		}
+
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to save rule set to file on node %s: %w", node.Name, err)
+		}
+
+		installCmd := []string{"bash", "-c", "yum update -y && yum install -y nftables"}
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, installCmd)
+		if err != nil {
+			return fmt.Errorf("failed to install nftables on  debugpod on node%s: %w", node.Name, err)
+		}
+
+		command = []string{
+			"bash", "-c", "/usr/sbin/nft -f nft/firewall.nft",
+		}
+
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to apply rule set %q on node %s: %w", command, node.Name, err)
+		}
+
+		// Check the output of nft list ruleset
+		command = []string{"bash", "-c", "/usr/sbin/nft list ruleset"}
+		output, err := utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to list NFT ruleset on node %s: %w", node.Name, err)
+		}
+
+		if len(output) == 0 {
+			return fmt.Errorf("no output from 'nft list ruleset' on node %s: ", node.Name)
+		}
+		command = []string{
+			"bash", "-c",
+			"/usr/sbin/nft list ruleset > /etc/nftables.conf",
+		}
+
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to save NFT ruleset to file on node %s: %w", node.Name, err)
+		}
+
+		command = []string{"bash", "-c",
+			"systemctl enable nftables",
+		}
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to enable nftables %q on node %s: %w", command, node.Name, err)
+		}
+
+		artifactsDir, ok := os.LookupEnv("ARTIFACT_DIR")
+		if !ok {
+			fmt.Print("env var ARTIFACT_DIR is not set")
+		}
+		comArtifactsDir := filepath.Join(artifactsDir, "commatrix")
+		err = os.MkdirAll(comArtifactsDir, 0755)
+		if err != nil {
+			return fmt.Errorf("failed to create artifactsDir %w", err)
+		}
+
+		// save the nft ruleset on ArtifactDirPath.
+		err = os.WriteFile(filepath.Join(comArtifactsDir, fmt.Sprintf("nftables-%s.nft", node.Name)), output, 0644)
+		if err != nil {
+			return fmt.Errorf("failed to save nft file on for node artifacts %s %w", node.Name, err)
+		}
+	}
```

Comment:
this may be in a different function + as this is a test please add some test like checking rules are applied or something like that